cli/src/main/java/org/jboss/as/cli/impl/ModelControllerClientFactory.java

```diff
@@ -50,39 +51,19 @@
         void handleClose();
     }
 
-    ModelControllerClient getClient(ControllerAddress address, CallbackHandler handler,
+    ModelControllerClient getClient(ControllerAddress address, CallbackHandler handler, Set<CallbackKind> callbackKinds,
             boolean disableLocalAuth, SecurityFactory<SSLContext> sslContextFactory, boolean fallbackSslContext, int connectionTimeout,
             ConnectionCloseHandler closeHandler, ProtocolTimeoutHandler timeoutHandler, String clientBindAddress) throws IOException;
 
-    ModelControllerClientFactory DEFAULT = new ModelControllerClientFactory() {
```

Comment:
@honza889, now that you have added support for the Builder, you should possibly restore the DEFAULT instance and update the call to the Builder with the Kinds. We could create a separate JIRA to handle dead code.